pd/src/app.rs

```diff
@@ -195,27 +204,30 @@ impl App {
     ) -> impl Future<Output = Result<Response, BoxError>> {
         let state = self.state.clone();
         let mempool_nullifiers = self.mempool_nullifiers.clone();
-        let nct_root = self.note_commitment_tree.root2();
-        let finished_signal = self.completion_tracker.start();
+        let recent_anchors = self.recent_anchors.clone();
```

Comment:
Why are we cloning the `VecDeque` of recent anchors here? Should there be a shared reference instead?

Reply:
It's cloning just because it's the simplest way to access the anchors from inside the future -- we could use a shared reference by wrapping it in an `Arc` or something, but then we'd have to think about how we update the data, etc.  Not claiming that this is the best solution, but it works just well enough while we're testing to avoid transactions getting stuck because the state rolled forward while they were being submitted.

Reply:
That seems okay to me, then, provided we revisit later. I think an `Arc<RwLock<VecDeque<...>>>` should work well here; we don't need the futures-aware `RwLock` because we're not going to need to lock across an `.await`-point.